src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -545,6 +545,26 @@ protected DynamoModel(IStartConfiguration config)
             LogWarningMessageEvents.LogWarningMessage += LogWarningMessage;
         }
 
+        private void EngineController_TraceReconcliationComplete(TraceReconciliationEventArgs obj)
+        {
+            Debug.WriteLine("TRACE RECONCILIATION: {0} serializables were orphaned.", obj.OrphanedSerializables.Count());
+
+            var orphans = new List<ISerializable>();
+            foreach (HomeWorkspaceModel ws in Workspaces.Where(w => w is HomeWorkspaceModel))
```

Comment:
Some of the orphan finding logic had to stay in `DynamoModel`. Specifically, the logic to find nodes in the workspace that are not represented in the `CallSiteCache`, in the case where a node is deleted before the graph is updated. This logic needs to be in `DynamoModel` because `RunTimeData` doesn't know anything about the current workspace.